crawler-cli/src/main/java/org/gbif/crawler/pipelines/dwca/DwCAToAvroCommandVerification.java

```diff
@@ -0,0 +1,153 @@
+    return new DwCAToAvroCommandVerification(configuration);
+  }
+
+  private DwCAToAvroCommandVerification(DwCAToAvroConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  public DwCAToAvroCommandVerification with(DwcaValidationFinishedMessage receivedMessage) {
+    this.receivedMessage = receivedMessage;
+    return this;
+  }
+
+  /**
+   * verifies parameters and return calculated configuration parameters
+   *
+   * @return Calculated Configuration Parameters
+   */
+  public DwCA2AvroConfigurationParameter verifyParametersAndGetResourceConfigurations() {
+    /*
+      verifies the input configuration and messages are not null
+     */
+    Objects.requireNonNull(configuration, "Configuration cannot be null");
+    Objects.requireNonNull(receivedMessage, "Received message cannot be null");
+    /*
+      checks existence of archiveRepository provided in the yaml config
+     */
+    String inputBasePath = configuration.archiveRepository.endsWith(File.separator)
+      ? configuration.archiveRepository
+      : configuration.archiveRepository.concat(File.separator);
+    Assert.isLegal(Files.exists(new File(inputBasePath).toPath()),
+                   "Illegal Argument archiveRepository is not a valid path");
+    /*
+      calculates and checks existence of DwC Archive
+    */
+    String absoluteDwCAExportPath = inputBasePath.concat(this.receivedMessage.getDatasetUuid().toString());
+    String absoluteDwCAPath = absoluteDwCAExportPath.concat(".zip");
+    Assert.isLegal(Files.exists(new File(absoluteDwCAPath).toPath()),
+                   "Illegal Argument " + absoluteDwCAPath + " Input DwC Archive not available");
+
+    Configuration config = new Configuration();
+    config.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+    config.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+
+    URI fsURI = URI.create(configuration.exportAvroBaseURL);
+    FileSystem fs;
+    try {
+      fs = FileSystem.get(fsURI, config);
+    } catch (IOException ex) {
+      throw new IllegalArgumentException("Cannot get a valid filesystem from provided uri "
+                                         + configuration.exportAvroBaseURL, ex);
+    }
+    /*
+    checks existence of provided path for exporting final avro file
+     */
+    String baseURL = configuration.exportAvroBaseURL.endsWith(File.separator)
+      ? configuration.exportAvroBaseURL
+      : configuration.exportAvroBaseURL + File.separator;
+
+    try {
+      Assert.isLegal(fs.exists(new Path(baseURL)), "export avro url provided donot exists or is invalid");
+    } catch (IOException e) {
+      throw new IllegalArgumentException("file system or path has error, " + e.getMessage());
+
+    }
+
+    Path absoluteAvroExportPath = new Path(baseURL
+                                           + "data"
+                                           + File.separator
```

Comment:
us hadoop.fs.Path.SEPARATOR instead of the local file system separator

Reply:
done